### PR TITLE
[Ice] add supply to createManyFromApi

### PR DESCRIPTION
**Pull request by icegriffinguru**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ export function parseDataNft(value: NftType): DataNft {
     royalties: value.royalties / 100,
     nonce: value.nonce,
     collection: value.collection,
+    balance: value.balance ? Number(value.balance) : 0,
```
Comment (bucurdavid) [inline src/utils.ts:53]: It shouldn't be supply instead of balance ? @icegriffinguru

**Review by bucurdavid — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 1 above)
Comment (newbreedofgeek) [reply src/utils.ts:53]: @icegriffinguru @bucurdavid yes, we should use terminology that's the same everywhere. In the Data DEX, where / what is this attribute called in the back end/ api?

image

Comment (icegriffinguru) [reply src/utils.ts:53]: `balance` and `supply` are different terms.
`balance` is what you are owning and `supply` is total number of issued amount of that nonce. @bucurdavid

File: src/datanft.ts [hunk 2]
```diff
@@ -90,7 +91,7 @@ export class DataNft {
       createNftIdentifier(tokenIdentifier, nonce)
     );
     const response = await fetch(
-      `${this.apiConfiguration}/nfts?identifiers=${identifiers.join(',')}`
+      `${this.apiConfiguration}/nfts?identifiers=${identifiers.join(',')}&withSupply=true`
```
Comment (bucurdavid) [inline src/datanft.ts:94]: By adding ```&withSupply=true```,  it should parse it correctly. ```  supply: value.supply ? Number(value.supply) : 0, ```

**Review by bucurdavid — APPROVED** [review]

**Pull request merged**